ADR/0011-roles-and-permissions.md

```diff
@@ -0,0 +1,81 @@
+|               | Pipeline Results        | results.tekton.dev        | get, list                               | results, records
+|               | IntegrationTestScenario | appstudio.redhat.com      | get, list, watch                        | integrationtestscenarios
+|               | Enterprise contract     | appstudio.redhat.com      | get, list, watch                        | enterprisecontractpolicies
+|               | Release Strategy        | appstudio.redhat.com      | get, list, watch                        | releases, releasestrategies, releaseplans
+|               | Release Admission Plan  | appstudio.redhat.com      | get, list, watch                        | releaseplanadmissions
+|               | *JVM Build Service*     | jvmbuildservice.io        | get, list, watch                        | jbsconfigs, artifactbuilds
+|               | *Service Access*        | appstudio.redhat.com      | get, list, watch                        | spiaccesstokenbindings, spiaccesschecks, spiaccesstokens, spifilecontentrequest
+|               | *Configs*               |                           | get, list, watch                        | configmaps
+|               | *Secrets*               |                           |                                         | secrets
+|               | Add User                |
+|               | User group (with SSO)   |
+| Maintainer    | Workspace               | Access to namespaces that backs workspace                           |
+|               | Application & Component | appstudio.redhat.com      | get, list, watch, create, update, patch | applications, components, componentdetectionqueries
+|               | Environment             | appstudio.redhat.com      | get, list, watch                        | promotionruns, snapshotenvironmentbindings, snapshots, environments
+|               | *GitOps*                | managed-gitops.redhat.com | get, list, watch                        | gitopsdeployments
+|               | PipelineRun             | tekton.dev                | create, get, list, watch                | pipelineruns
+|               | Pipeline Results        | results.tekton.dev        | get, list                               | results, records
+|               | IntegrationTestScenario | appstudio.redhat.com      | *                                       | integrationtestscenarios
+|               | Enterprise contract     | appstudio.redhat.com      | get, list, watch                        | enterprisecontractpolicies
+|               | Release Strategy        | appstudio.redhat.com      | *                                       | releases, releasestrategies, releaseplans
+|               | Release Admission Plan  | appstudio.redhat.com      | *                                       | releaseplanadmissions
+|               | *JVM Build Service*     | jvmbuildservice.io        | get, list, watch                        | jbsconfigs, artifactbuilds
```

Comment:
I think the maintainer role should be able to create/update/patch the JBSConfig. This is how they "turn on" java builds for themselves.

@stuartwdouglas can you confirm?

See also:

* https://issues.redhat.com/browse/MGDSR-546 for RHOSR onboarding instructions
* https://issues.redhat.com/browse/STONE-217 unscoped, for the future, enhancements to HAC to permit users to toggle this on and off.

Reply:
Yes, definitely

Reply:
They should also be able to annotate ArtifactBuild objects, as this is used to trigger rebuilds.

Reply:
Updated for the create/update